brainiak/reprsimil/brsa.py

```diff
@@ -87,106 +227,220 @@ class BRSA(BaseEstimator):
         The other way is to take the first n_nureg principal components
         in the residual after subtracting the response to the design matrix
         from the data, and use these components as the nuisance regressor.
-        If this flag is turned on, the nuisance regressor provided by the
-        user is used only in the first round of fitting. The PCs from
-        residuals will be used in the next round of fitting.
+        This flag is for the second approach. If turned on,
+        PCA or factor analysis will be applied to the residuals
+        to obtain new nuisance regressors in each round of fitting.
+        These two approaches can be combined. If the users provide nuisance
+        regressors and set this flag as True, then the first few principals
+        of the residuals after subtracting both the responses to design
+        matrix and nuisance regressors will be used in addition to the
+        nuisance regressors provided by the users.
         Note that nuisance regressor is not required from user. If it is
-        not provided, DC components for each run will be used as nuisance
-        regressor in the initial fitting.
-    n_nureg: int, default: 6
+        not provided, DC components for each run will be included as nuisance
+        regressor regardless of the auto_nuisance parameter.
+    n_nureg: int or string. Default: 'BIC'
         Number of nuisance regressors to use in order to model signals
         shared across voxels not captured by the design matrix.
-        This parameter will not be effective in the first round of fitting.
-    GP_space: boolean, default: False
+        This number is in addition to any nuisance regressor that the user
+        has already provided.
+        If set to 'BIC', the number of nuisance regressor will be
+        automatically determined based on BIC metric
+        of probabalistic PCA model. (Minka 2000, NIPS)
+    nureg_method: string, 'PCA', 'FA', 'ICA' or 'SPCA'. Default: 'ICA'
```

Comment:
For any way of choosing number of components, say the one we keep now, the number makes sense for ICA because PCA is usually the first step of ICA. The assumption for PCA might not directly apply to FA. I am not sure about SPCA. But practically I find the result of SPCA is very close to PCA. I do not know if there are other ways to determine the number of components for FA. But I want to leave this option available in case people want to try.